### PR TITLE
Add Signup page

### PR DESCRIPTION
**Pull request by Sahar-AbdelSamad**
# Hi @piero-vic    👋🏻 

Thanks for taking the time to review the project. Here are some things important to note about it:

### Changes implemented in this branch 

✅  Created the Signup page
✅  Imported it to `App.js`

## Final notes

Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -16,7 +17,7 @@ function App() {
         <div className="flex-1">
           <Routes>
             <Route path="/login" element={<Login loggedIn={loggedIn} />} />
-            <Route path="/signup" element={<h1>Sign Up Page</h1>} />
+            <Route path="/signup" element={<Signup />} />
```
Comment (piero-vic) [inline src/App.js]: Good job so far. I think you could also pass `loggedIn` to the `Signup` component. See my other comment to know why.

File: src/pages/Signup.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { signupUser } from '../redux/actions/auth';
+
+const Signup = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { register, handleSubmit } = useForm();
+  const onFormSubmit = (data) => {
+    dispatch(signupUser(data));
+    navigate('/');
+  };
```
Comment (piero-vic) [inline src/pages/Signup.js]: When the form is submitted you're dispatching an action and then using the navigation. However this actions needs to interact with the API so is asynchronous. The navigate function will run before we can signup the user. 

You can see in [this video](https://www.loom.com/share/0549137472f74279b357dddfc407c74e) how we are redirected to the login page for a fraction of a second after the form submit. Please refer to the code on the Login component to fix this behavior.

**Review by piero-vic — DISMISSED** [review]
## Status: Changes Required :recycle: 

Hi @Sahar-AbdelSamad

We're almost done with this. There's only one little problem. Read the comments for more details. :+1:

**Review by Reem-lab — APPROVED** [review]
Hi @Sahar-AbdelSamad ,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯
  - Amazing design
  
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

File: src/pages/Signup.js (hunk 2 above)
Comment (Sahar-AbdelSamad) [reply src/pages/Signup.js]: Thank you for the reat explanation😀👍

**Pull request merged**